Future - Drop Emotion 10 types in lib/theming

ndelangen opened the pull request.

Andarist committed:
Drop Emotion 10 types

ndelangen committed:
Merge branch 'future/base' into drop-emotion-10-types

# Conflicts:
#	lib/theming/src/emotion10types.ts

ndelangen committed:
Merge branch 'future/tsup-libs-1' into drop-emotion-10-types

# Conflicts:
#	docs/snippets/html/button-story-with-args.ts.mdx

ndelangen committed:
Merge branch 'future/tsup-libs-1' into drop-emotion-10-types

ndelangen committed:
Merge branch 'future/pbm/stage0' into drop-emotion-10-types

ndelangen committed:
use TSup for lib/theming

ndelangen commented:
@Andarist OK, this is getting into shape now, I think.

I think it's prepped for a proper integration now. But the outputted types aren;t correct. We're getting the empty Theme interface in our components now, not the Theme interface we define.

![Screen Shot 2022-06-29 at 10 10 39](https://user-images.githubusercontent.com/3070389/176386023-cfc79823-8c5b-4b1f-8d87-3bcddeb41fad.png)

![Screen Shot 2022-06-29 at 10 10 58](https://user-images.githubusercontent.com/3070389/176386086-c170d0ed-649d-411b-b2dd-be729287a20c.png)

We get this one:
![Screen Shot 2022-06-29 at 10 13 26](https://user-images.githubusercontent.com/3070389/176386586-5c235ce9-6e01-4d08-b3f3-4dcfeddfdfdb.png)

Not this one 😢 
![Screen Shot 2022-06-29 at 10 13 46](https://user-images.githubusercontent.com/3070389/176386653-e2f8b724-3f99-4af9-a1e3-699b23b3bdaa.png)

ndelangen committed:
Merge branch 'future/pbm/stage0' into drop-emotion-10-types

# Conflicts:
#	lib/theming/package.json

ndelangen committed:
Merge branch 'future/pbm/stage0' into drop-emotion-10-types

# Conflicts:
#	lib/theming/package.json

ndelangen committed:
Merge branch 'future/pbm/stage0' into drop-emotion-10-types

ndelangen committed:
Merge branch 'future/pbm/stage0' into drop-emotion-10-types

ndelangen committed:
Merge branch 'future/pbm/stage0' into drop-emotion-10-types

ndelangen committed:
Merge branch 'future/pbm/stage0' into drop-emotion-10-types

ndelangen committed:
Merge branch 'future/pbm/stage0' into drop-emotion-10-types

ndelangen committed:
Merge branch 'future/pbm/stage0' into drop-emotion-10-types

ndelangen committed:
Merge branch 'future/pbm/stage0' into drop-emotion-10-types

ndelangen committed:
Merge branch 'future/prebuilt-manager-esbuild' into future/drop-emotion-10-types

Andarist committed:
Make Emotion's `Theme` augmentation work in prod and dev

Andarist commented:
I believe that I got it working - both for prod bundles and for the dev environment. Please retest if it matches your expectations.

Andarist commented on lib/theming/src/convert.ts:
```diff
@@ -1,7 +1,6 @@
-
 import { background, typography, color } from './base';
-import { Theme, Color, ThemeVars } from './types';
+import { StorybookTheme, Color, ThemeVars } from './types';
```
note that within the `theming` package you have to reference `StorybookTheme` and not `Theme` because `Theme` doesn't exist yet - it's only added after bundling and into the reexporting file

Andarist commented on lib/theming/src/emotionAugmentation.d.ts:
```diff
@@ -0,0 +1,9 @@
+// this file is only actually used in development
```
Believe it or not but this has to be in a separate file and I can't reuse `src/typings.d.ts` - I've already stumbled upon a problem like this in the past, TS rules are weird when it comes to those `d.ts` files and I don't fully understand them. When put in the other file I couldn't use `import` at the top-level and without it, the tsup didn't recognize other Emotion exports (such as `keyframes`, `css` etc). This way I only touch what I actually want and leave the rest as-is

ndelangen commented:
I'll take a look tomorrow!

ndelangen committed:
Merge branch 'future/base' into future/drop-emotion-10-types

# Conflicts:
#	scripts/prepare/bundle.ts

ndelangen committed:
a bunch of fixes

ndelangen committed:
cleanup

ndelangen committed:
cleanup

ndelangen committed:
cleanup

ndelangen commented:
- Merged in `future/base`
- Fixed the build
- Cleanup up all usage of `<{}>` that served no purpose
- Cleanup up use of `FunctionComponent` and replaced it with `FC` which is shorter
- Extracted a few components wrapped directly with `styled` so their types are working properly

ndelangen commented:
@shilman can we take a look at the doc blocks UI changes being detected?

Som flacky styling (there's a underline underneath the "raw" text.

ndelangen committed:
Merge branch 'future/base' into future/drop-emotion-10-types

# Conflicts:
#	lib/blocks/src/blocks/DocsContainer.tsx
#	lib/blocks/src/blocks/SourceContainer.tsx
#	lib/blocks/src/blocks/Subtitle.tsx
#	lib/blocks/src/blocks/Title.tsx
#	lib/blocks/src/components/ColorPalette.tsx
#	lib/blocks/src/components/IconGallery.tsx
#	lib/blocks/src/components/Source.tsx
#	lib/blocks/src/components/Story.tsx
#	lib/components/src/placeholder/placeholder.tsx
#	lib/ui/src/components/preview/toolbar.tsx
#	lib/ui/src/components/preview/utils/types.tsx
#	lib/ui/src/components/sidebar/Sidebar.tsx
#	lib/ui/src/components/sidebar/TreeNode.tsx

ndelangen committed:
cleanup

ndelangen committed:
fix

ndelangen commented:
@shilman can you assist figuring out why there's a underline in the `raw` button now? It's the only thing blocking this PR. 🙏

ndelangen committed:
fix
